src/port/port.tcl

```diff
@@ -3927,6 +3927,128 @@ proc action_echo { action portlist opts } {
     return 0
 }
 
+proc prompt { message default variable } {
+    puts -nonewline $message
+    flush stdout
+    gets stdin variable
+    if {[string trimleft $variable] eq ""} {
+        set variable $default
+    }
+    return 0
+}
+
+proc offer_clone {} {
+    global macports::os_platform macports::os_major macports::os_arch \
+        macports::sources macports::autoconf::rsync_path \
+        macports:rsync_options macports::getindex macports::sources_default
+
+    set msg4 {Rsyncing a PortIndex file...}
+    set msg5 {}
+
+    # OFFER GIT-BASED SOURCE AND ASK FOR PATH
+    set msg1 "You do not seem have a working copy of the macports-ports git repository, which is required to contribute this change."
+    set retvalue [$macports::ui_options(questions_yesno) $msg1 "clone" {} {y} 0 "Create one now?"]
+    if {$retvalue == 1} {
+        # quit as user answered 'no'
+        return 0
+    }
+
+    set path [$macports::ui_options(questions_input) "Where do you want to create your working copy?" [file join [file normalize ~] macports-ports] 0]
+
+    # WALK DOWN FROM PATH TO ROOT AND ASK FOR PERMISSIONS IF NEEDED
+    set paths {}
+    set temp $path
+    while {$temp != [file normalize /]} {
+        ui_msg "$temp [file readable $temp]"
+        if {[file readable $temp]} {
```

Comment:
I think we should test whether the other permission bits contain the "executable" bit instead, because that would be independent of which user we're running this under, while `file readable` isn't.

Let's use this:
```tcl
expr {[string index [file attributes /Users/clemens -permissions] end] & 1}
```